src/models/document/document-content.ts

```diff
@@ -295,7 +300,7 @@ export const DocumentContentModel = types
           row.removeTileFromRow(tileId);
         }
         // track empty rows
-        if (row.tiles.length === 0) {
+        if (row.tiles.length === 0 && !row.isSectionHeader) {
```

Comment:
It occurs to me that rather than exposing this kind of implementation detail to clients it would be better to add an `isEmpty()` accessor to the `RowModel` and then clients like this can just test `row.isEmpty`.